nemo/collections/nlp/data/language_modeling/megatron/gpt_prompt_tuning_dataset.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Prompt tuning dataset
+Expects data to be in the format:
+{"prompt_tag": "tag1", "text": "example text1"}
+{"prompt_tag": "tag1", "text": "example text2"}
+{"prompt_tag": "tag2", "text": "example text3"}
+
+"""
+import json
+
+import torch
+from tqdm import tqdm
+
+from nemo.core import Dataset
+from nemo.utils import logging
+
+__all__ = ["GPTPromptTuningDataset"]
+
+
+class GPTPromptTuningDataset(Dataset):
+    def __init__(
+        self,
+        dataset_path,
+        tokenizer,
+        num_prompt_tokens: int,
+        max_seq_length: int,
+        min_seq_length: int = 1,
+        add_bos_eos: bool = True,
+    ):
+        self.tokenizer = tokenizer
+        self.add_bos_eos = add_bos_eos
+        self.max_seq_length = max_seq_length
+        self.min_seq_length = min_seq_length
+        self.num_prompt_tokens = num_prompt_tokens
+        self.max_sent_length = max_seq_length - num_prompt_tokens
+        self.prompt_tags = []
```

Comment:
please add asserts to make sure initialization params make sense. E.g. min<max, max>0, and there are enough for num_prompt_tokens, etc.